pkg/asset/installconfig/azure/session.go

```diff
@@ -0,0 +1,155 @@
+package azure
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/Azure/go-autorest/autorest"
+	azureenv "github.com/Azure/go-autorest/autorest/azure"
+	"github.com/Azure/go-autorest/autorest/azure/auth"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"gopkg.in/AlecAivazis/survey.v1"
+)
+
+var authFileLocation = os.Getenv("HOME") + "/.azure/osServicePrincipal.json"
+
+//Session is an object representing session for subscription
+type Session struct {
+	SubscriptionID string
+	Authorizer     autorest.Authorizer
+}
+
+// GetSession returns an azure session by checking credentials
+// and, if no creds are found, asks for them and stores them on disk in a config file
+func GetSession() (*Session, error) {
```

Comment:
this should be reversed:
```
<try get credentials from env> https://github.com/Azure/go-autorest/blob/master/autorest/azure/auth/auth.go#L64
<try get credentials from file> https://github.com/Azure/go-autorest/blob/master/autorest/azure/auth/auth.go#L226

<if both failed>
  <ask for creds>
  <setup the credentials on host so that "get credentials from file" succeeds>
```

Reply:
it's not getting them from ENV.
I set the env variable that specifies the file location, so that the authorizer uses this instead of the default azure one.
once it is there, it won't ask for credentials, and use that instead.

- set env file location
- if it's there, use it
- if not, ask for creds

Reply:
Resolving as won't change in this iteration